s3/utils.bal

```diff
@@ -407,3 +423,30 @@ isolated function handleHttpResponse(http:Response httpResponse) returns @tainte
         return error(xmlPayload.toString());
     }
 }
+
+isolated function getIAMCredentials() returns IAMCredentials|error {
+    // Create HTTP client for metadata service
+    http:Client metadataClient = check new (METADATA_TOKEN_URL);
```

Comment:
Can't we have these clients as global variables?

Reply:
AFAIU this is required only at the application startup. Hence, we don't need to preserve this client for future use (or in a global context).

Reply:
This is only required for a specific scenario (IAM Role-based auth).